File: LvlUpBack-main/src/main/java/com/levelup/gaming/models/Event.java
package com.levelup.gaming.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Event {
    @Id
    private String id;
    private String title;
    private LocalDate date;
    private LocalTime time;
    private String location;
    @Column(length = 1000)
    private String mapEmbed;
    private String mapUrl;
    private String notes;
}

File: LvlUpFront-main/src/types/Event.ts


export interface Event {
    id: string;
    title: string;
    date: string; 
    time: string; 
    location: string;
    mapEmbed: string;
    notes?: string;
}

export interface EventFormData {
    title?: string;
    date?: string;
    time?: string;
    location?: string;
    mapEmbed?: string;
    notes?: string;
}

export interface StatusMessage {
    msg: string;
    type: 'success' | 'danger';
}
